internal/bminventory/inventory.go

```diff
@@ -2363,6 +2363,18 @@ func (b *bareMetalInventory) integrateWithAMSClusterUpdateName(ctx context.Conte
 func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]interface{}, cluster *common.Cluster, params installer.V2UpdateClusterParams, log logrus.FieldLogger, interactivity Interactivity) error {
 	apiVip := cluster.APIVip
 	ingressVip := cluster.IngressVip
+
+	// We are checking if the cluster is requested to be a dual-stack cluster based on the Cluster
+	// Networks provided.
+	//
+	// TODO(mko) As updateNonDhcpNetworkParams is called before updateNetworks, this check looks
+	//           at the already configured networks and not the ones inside V2UpdateClusterParams.
+	//           An extension would be to check if V2UpdateClusterParams configures any of those
+	//           and if that's the case, instead of GetConfiguredAddressFamilies(cluster) we
+	//           should call something like GetRequestedAddressFamilies(cluster).
+	reqV4, reqV6, _ := network.GetConfiguredAddressFamilies(cluster)
+	reqDualStack := reqV4 && reqV6
```

Comment:
Even though `reqDualStack` could be TRUE if `reqV4` and `reqV6` are both FALSE, we always handle this variable together with the length of respective *-Networks variable so this is not causing an unexpected behaviour (because for non-empty networks we will always have either V4 or V6 or both present)

Reply:
Done